portfolio/src/test/java/com/google/sps/DataServletTest.java

```diff
@@ -0,0 +1,273 @@
+    comment.setProperty(dataServlet.TIMESTAMP_COLUMN_NAME, timestamp);
+    ds.put(comment);
+  }
+
+  @Test 
+  public void testPostSingleComment() throws Exception {
+    assertThat(getNumberOfEntiresInDatastore()).isEqualTo(0);
+
+    when(postRequest.getParameter(COMMENT_PARAMETER)).thenReturn(COMMENT_ONE);
+    dataServlet.doPost(postRequest, postResponse);
+    
+    assertThat(getNumberOfEntiresInDatastore()).isEqualTo(1);
+  }
+
+  @Test 
+  public void testPostMultipleComments() throws Exception {
+    assertThat(getNumberOfEntiresInDatastore()).isEqualTo(0);
+
+    when(postRequest.getParameter(COMMENT_PARAMETER)).thenReturn(COMMENT_ONE);
+    dataServlet.doPost(postRequest, postResponse);
+    when(postRequest.getParameter(COMMENT_PARAMETER)).thenReturn(COMMENT_TWO);
+    dataServlet.doPost(postRequest, postResponse);
+    when(postRequest.getParameter(COMMENT_PARAMETER)).thenReturn(COMMENT_THREE);
+    dataServlet.doPost(postRequest, postResponse);
+    
+    assertThat(getNumberOfEntiresInDatastore()).isEqualTo(3);
+  }
+
+  @Test 
+  public void testPostNull() throws Exception {
+    assertThat(getNumberOfEntiresInDatastore()).isEqualTo(0);
+
+    when(postRequest.getParameter(COMMENT_PARAMETER)).thenReturn(null);
+    dataServlet.doPost(postRequest, postResponse);
+    
+    assertThat(getNumberOfEntiresInDatastore()).isEqualTo(0);
+  }
+
+  @Test
+  public void testGetSingleComment() throws Exception {
+    addEntityToDatastore(COMMENT_ONE, TIMESTAMP_ONE);
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter printWriter = new PrintWriter(stringWriter);
+    when(getResponse.getWriter()).thenReturn(printWriter);
+
+    dataServlet.doGet(getRequest, getResponse);
+
+    printWriter.flush(); //may not have flushed yet
+    String[] comments = convertStringToArray(stringWriter.toString());
+    assertThat(comments[0]).isEqualTo(COMMENT_ONE);
+  }
+  
+  @Test
+  public void testGetSomeComments() throws Exception {
+    int NUM_COMMENTS_TO_DISPLAY = 2;
+
+    addEntityToDatastore(COMMENT_ONE, TIMESTAMP_ONE);
+    addEntityToDatastore(COMMENT_TWO, TIMESTAMP_TWO);
+    addEntityToDatastore(COMMENT_THREE, TIMESTAMP_THREE);
+    addEntityToDatastore(COMMENT_FOUR, TIMESTAMP_FOUR);
+    addEntityToDatastore(COMMENT_FIVE, TIMESTAMP_FIVE);
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter printWriter = new PrintWriter(stringWriter);
+    when(getResponse.getWriter()).thenReturn(printWriter);
+    when(getRequest.getParameter(NUMBER_COMMENTS_DISPLAYED_PARAMETER)).thenReturn(Integer.toString(NUM_COMMENTS_TO_DISPLAY));
+    dataServlet.doGet(getRequest, getResponse);
+
+    printWriter.flush(); //may not have flushed yet
+    String[] comments = convertStringToArray(stringWriter.toString());
+    assertThat(comments[0]).isEqualTo(COMMENT_ONE);
+    assertThat(comments[1]).isEqualTo(COMMENT_TWO);  
+    assertThat(comments.length).isEqualTo(NUM_COMMENTS_TO_DISPLAY);
```

Comment:
If you wanna be fancy
```suggestion
    assertThat(comments).hasLength(NUM_COMMENTS_TO_DISPLAY);
```

Reply:
ah that's great! I'll keep that in mind when I use that assertion scheme next!